code/modules/antagonists/nukeop/equipment/nuclear_bomb/beer_nuke.dm

```diff
@@ -59,5 +61,10 @@
 
 /obj/machinery/nuclearbomb/beer/really_actually_explode(detonation_status)
 	disarm_nuke()
-	var/datum/round_event_control/event = locate(/datum/round_event_control/scrubber_overflow/beer) in SSevents.control
-	event.runEvent()
+	if (flood_reagent != initial(flood_reagent))
+		var/datum/round_event_control/scrubber_overflow/custom/event = locate(/datum/round_event_control/scrubber_overflow/custom) in SSevents.control
+		event.custom_reagent = flood_reagent
+		event.runEvent()
+	else
+		var/datum/round_event_control/event = locate(/datum/round_event_control/scrubber_overflow/beer) in SSevents.control
+		event.runEvent()
```

Comment:
What's the point in having 2 separate events? Beer can just run off of custom using beer as the custom reagent, can't it?